sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c

```diff
@@ -125,9 +119,6 @@ bool test_main(void) {
   if (rstmgr_reset_info == kDifRstmgrResetInfoPor) {
     LOG_INFO("POR reset");
 
-    LOG_INFO("Scrambling...");
-    sram_ctrl_testutils_scramble(&ret_sram);
```

Comment:
@weicaiyang this has actually been added on purpose to check the scrambling capability of the retention SRAM (action item from here: https://github.com/lowRISC/opentitan/issues/13543).

But thinking about this more, we may have to update the testplan and expand this test with another scenario to cover all cases. I.e., we basically want to make sure the following four cases are tested:
1. light sleep, no scrambling -> data preserved
2. deep sleep, no scrambling -> data preserved
3. light sleep, with scrambling -> data preserved
4. deep sleep, with scrambling -> data NOT preserved

Does this make sense?

Reply:
This is more comprehensive. Let me update the test and testplan.

Reply:
After having a close look with @matutem, low-power exit (regardless of light or deep sleep) doesn't reset ret_sram as it's in AON domain. Wake-up from deep sleep only resets the non-AON domain. We need a real reset request in order to reset the ret_sram. (the reset request doesn't need to be in a sleep mode.)

So we actually have 6 cases to cover.
1. light sleep, no scrambling -> data preserved
2. deep sleep, no scrambling -> data preserved
3. reset due to a reset request, no scrambling -> data preserved
4. light sleep, with scrambling -> data preserved
5. deep sleep, with scrambling -> ~data NOT preserved~ data preserved
6. reset due to a reset request, with scrambling -> data NOT preserved

WDTY? @msfschaffner

Reply:
Yeah that makes sense - thanks for double checking the reset behavior!

Reply:
@msfschaffner Thanks for the confirmation. Will update the test accordingly.

BTW, do you know if we recommend to enable scrambling for ret_sram in the real product, or we prefer to have the data preserved across reset as well?

Reply:
That is ultimately a SW decision, but I believe the current guidance is to not request a key so that data is preserved across resets.

CC @tjaychen

Reply:
yeah that's right for retention ram.
I think retention ram is just relying on the original key for fault protection.

Reply:
Make sense. Thanks @msfschaffner @tjaychen